contrib/dimfeld/httptreemux/v5/option.go

```diff
@@ -0,0 +1,85 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package httptreemux provides functions to trace the dimfeld/httptreemux/v5 package (https://github.com/dimfeld/httptreemux).
+package httptreemux
+
+import (
+	"math"
+	"net/http"
+
+	"github.com/dimfeld/httptreemux/v5"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+)
+
+type routerConfig struct {
+	serviceName   string
+	spanOpts      []ddtrace.StartSpanOption
+	analyticsRate float64
+	resourceNamer func(*httptreemux.TreeMux, http.ResponseWriter, *http.Request) string
+}
+
+// RouterOption represents an option that can be passed to New.
+type RouterOption func(*routerConfig)
+
+func defaults(cfg *routerConfig) {
+	if internal.BoolEnv("DD_TRACE_HTTPTREEMUX_ANALYTICS_ENABLED", false) {
+		cfg.analyticsRate = 1.0
+	} else {
+		cfg.analyticsRate = globalconfig.AnalyticsRate()
+	}
+	cfg.serviceName = "http.router"
+	if svc := globalconfig.ServiceName(); svc != "" {
+		cfg.serviceName = svc
+	}
+	cfg.resourceNamer = defaultResourceNamer
+}
+
+// WithServiceName sets the given service name for the returned router.
+func WithServiceName(name string) RouterOption {
+	return func(cfg *routerConfig) {
+		cfg.serviceName = name
+	}
+}
+
+// WithSpanOptions applies the given set of options to the span started by the router.
+func WithSpanOptions(opts ...ddtrace.StartSpanOption) RouterOption {
+	return func(cfg *routerConfig) {
+		cfg.spanOpts = opts
+	}
+}
+
+// WithAnalytics enables Trace Analytics for all started spans.
+func WithAnalytics(on bool) RouterOption {
+	return func(cfg *routerConfig) {
+		if on {
+			cfg.analyticsRate = 1.0
+		} else {
+			cfg.analyticsRate = math.NaN()
+		}
+	}
+}
+
+// WithAnalyticsRate sets the sampling rate for Trace Analytics events
+// correlated to started spans.
+func WithAnalyticsRate(rate float64) RouterOption {
+	return func(cfg *routerConfig) {
+		if rate >= 0.0 && rate <= 1.0 {
+			cfg.analyticsRate = rate
+		} else {
+			cfg.analyticsRate = math.NaN()
+		}
+	}
+}
+
```

Comment:
All of this can be deleted

Reply:
- [ ] remove `WithAnalyticsRate` and usages